test/modules/home/home_view_test.dart

```diff
@@ -0,0 +1,239 @@
+          final boxDecoration = dot.decoration! as BoxDecoration;
+          final color = i == 0
+              ? HomeView.dotPageControlHighlightColor
+              : HomeView.dotPageControlNormalColor;
+          expect(boxDecoration.color, color);
+        }
+      });
+
+      context("when swipes right to left on screen", () {
+        beforeEach((tester) async {
+          final location = tester.getCenter(find.byKey(HomeView.bodyKey));
+          await tester.flingFrom(
+              location, Offset(-location.dx, 0), location.dx);
+          await tester.pumpAndSettle();
+        });
+
+        it("triggers carousel slider to show the second survey information",
+            (tester) async {
+          final survey = surveys[1];
+          expect(find.text(survey.title!), findsOneWidget);
+          expect(find.text(survey.description!), findsOneWidget);
+
+          final widget = tester
+              .widget<Image>(find.byKey(HomeView.backgroundImageSlideItemKey));
+          final networkImage = widget.image as NetworkImage;
+          expect(networkImage.url, survey.coverImageUrl);
+        });
+
+        it("triggers page control to highlight the second dot", (tester) async {
+          final dot = tester
+              .widgetList<Container>(find.byKey(HomeView.dotPageControlKey))
+              .toList()[1];
+          final boxDecoration = dot.decoration! as BoxDecoration;
+          expect(boxDecoration.color, HomeView.dotPageControlHighlightColor);
+        });
+      });
+
+      context("when swipes left to right on screen", () {
+        beforeEach((tester) async {
+          final location = tester.getCenter(find.byKey(HomeView.bodyKey));
+          await tester.flingFrom(location, Offset(location.dx, 0), location.dx);
+          await tester.pumpAndSettle();
+        });
+
+        it("triggers carousel slider to display the last survey information",
+            (tester) async {
+          final survey = surveys.last;
+          expect(find.text(survey.title!), findsOneWidget);
+          expect(find.text(survey.description!), findsOneWidget);
+
+          final widget = tester
+              .widget<Image>(find.byKey(HomeView.backgroundImageSlideItemKey));
+          final networkImage = widget.image as NetworkImage;
+          expect(networkImage.url, survey.coverImageUrl);
+        });
+
+        it("triggers page control to highlight the last dot", (tester) async {
+          final dot = tester
+              .widgetList<Container>(find.byKey(HomeView.dotPageControlKey))
+              .toList()
+              .last;
+          final boxDecoration = dot.decoration! as BoxDecoration;
+          expect(boxDecoration.color, HomeView.dotPageControlHighlightColor);
+        });
+      });
+    });
+
+    describe("its beginSkeletonAnimation() is called", () {
+      beforeEach((tester) async {
+        module.view.beginSkeletonAnimation();
+        await tester.pump();
+        await tester.pump();
```

Comment:
just a bit curious if we need to `pump()` 2 times? 🤔

Reply:
I end up with that solution when 
```dart
await tester.pump(Duration(seconds: 10));
```
won't work for me. 😓 

And, I'm not sure why this trick works so gotta go deeper into how `pump()` works 😢

Reply:
@sleepylee btw, do u have any insight into this `pump()`? 😁

Reply:
also, it seems like `pump` triggers only 1 frame, and another pump == another frame trigger to rebuild. I think there is 1 default state and another is to trigger the animation here somehow 🤔

Reply:
@sleepylee  this animation is looped, so `pumpAndSettle()` won't work 😢 
```
// pumpAndSettle
Repeatedly calls pump with the given duration until there are no longer any frames scheduled
```

> also, it seems like pump triggers only 1 frame, and another pump == another frame trigger to rebuild. I think there is 1 default state and another is to trigger the animation here somehow 🤔

It sounds a good reason lol